server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java

```diff
@@ -237,12 +221,6 @@ protected Iterator<ToXContent> customXContentChunks(ToXContent.Params params) {
                             if (indexStats.getState() != null) {
                                 builder.field("status", indexStats.getState().toString().toLowerCase(Locale.ROOT));
                             }
-                            if (indexStats.getTierPreference() != null) {
-                                builder.field("tier_preference", indexStats.getTierPreference());
-                            }
-                            if (indexStats.getCreationDate() != null) {
-                                builder.field("creation_date", indexStats.getCreationDate());
-                            }
```

Comment:
I believe this is ok, I don't think any clients could have been expecting these fields yet and they were optional anyway so we can remove them without ceremony.